aten/src/ATen/native/mps/operations/UnaryOps.mm

```diff
@@ -52,8 +52,13 @@ void unary_op(const Tensor& self, const Tensor& output, std::string op_name, Una
       cachedGraph = tmpCachedGraph->as<MPSUnaryCachedGraph>();
     }
 
-    Placeholder selfPlaceholder = Placeholder(cachedGraph->inputTensor_, self);
-    Placeholder outputPlaceholder = Placeholder(cachedGraph->outputTensor_, output);
+    bool gatherTensorData = true;
+    if (!output.is_contiguous() || output.is_view()) {
+      gatherTensorData = false;
+    }
+
+    Placeholder selfPlaceholder = Placeholder(cachedGraph->inputTensor_, self, /*mpsShape=*/nullptr, gatherTensorData);
```

Comment:
The title of this PR concerns the strided 'outputs' only. So why should we skip gather for strided 'inputs' in here too?

Reply:
If both the input and output is strided, it means that the memory is ordered the same in both of them.

E.g lets take for example this case:
```
x = torch.rand((256,10), device='mps')
x = x.permute(1,0)
x.exp()
```

In the current code, when it reaches `exp`, both input and output are strided. The following happens: 
1.  It will allocate new memory for `input` in Placeholder and do a gather, which transposes the `input`
2. Output is also strided here and we allocate new memory and never saving back into the original buffer (data is lost when gets back to user)
3. If we don't allocate new memory for output (e.g mark it as false in Placeholder and don't run the gather part as it is now), we'll still have to do an additional transpose for output in the end, since we did it for the input (when the tensor is printed, we'll transpose it again since it's strided). We could workaround this also by making the output contiguous, but that seems the more hacky way to do it.